src/probability_theory/probability_space.lean

```diff
@@ -0,0 +1,777 @@
+  (event_empty Ω).val = ∅ := rfl
+
+/- Every probability space has the empty event, corresponding to the empty set. -/
+@[simp]
+lemma event_empty_val_def2 {Ω : Type*} [probability_space Ω] :
+  (@has_emptyc.emptyc (event Ω) _).val = ∅ := rfl
+
+/- The probability of an empty event is zero. -/
+@[simp]
+lemma Pr_event_empty {Ω : Type*} [probability_space Ω] : Pr[event_empty Ω] = 0 :=
+begin
+  have A1 : ↑(Pr[event_empty Ω]) = (0 : ennreal),
+  rw event_prob_def,
+  apply volume.empty,
+  simp at A1,
+  apply A1
+end
+
+/- The probability of the empty set is zero. -/
+@[simp]
+lemma Pr_event_empty' {Ω : Type*} [probability_space Ω] : Pr[(∅ : event Ω)] = 0 := Pr_event_empty
+
+def event_const (Ω : Type*) [probability_space Ω] (P : Prop) : event Ω :=
+  { val := {ω : Ω|P},
+  property := measurable_set.const P,}
+
+@[simp]
+lemma event_const_val_def {Ω : Type*} [probability_space Ω] (P : Prop) :
+  (event_const _ P).val = {ω : Ω|P} := rfl
+
+lemma event_const_true_eq_univ {Ω : Type*} [probability_space Ω] (P : Prop) : P →
+(event_const _ P) = event_univ Ω :=
+begin
+  intro A1,
+  apply event.eq,
+  simp [A1],
+end
+
+lemma event_const_false_eq_empty {Ω : Type*} [probability_space Ω] (P : Prop) : ¬P →
+(event_const _ P) = event_empty Ω :=
+begin
+  intro A1,
+  apply event.eq,
+  simp [A1],
+end
+
+lemma Pr_event_const_true {Ω : Type*} [probability_space Ω] (P : Prop) : P →
+Pr[(event_const Ω P)] = 1 :=
+begin
+  intro A1,
+  rw event_const_true_eq_univ,
+  apply Pr_event_univ,
+  exact A1,
+end
+
+lemma Pr_event_const_false {Ω : Type*} [probability_space Ω] (P : Prop) : ¬P →
+Pr[(event_const Ω P)] = 0 :=
+begin
+  intro A1,
+  rw event_const_false_eq_empty,
+  apply Pr_event_empty,
+  exact A1,
+end
+
+/- The "and" of two events is their intersection. -/
+def measurable_inter {Ω : Type*} [measurable_space Ω] (A B : event Ω) : event Ω :=
+  { val :=A.val ∩ B.val,
+  property := measurable_set.inter A.property B.property,}
+
+@[simp]
+lemma measurable_inter_val_def {Ω : Type*} [measurable_space Ω] (A B : event Ω) :
+  (measurable_inter A B).val= A.val ∩ B.val := rfl
+
+@[simp]
+lemma measurable_inter_val_def2 {Ω : Type*} [measurable_space Ω] (A B : event Ω) :
+  (A ∩ B).val= A.val ∩ B.val := rfl
+
+/- The weight of the eand of two events is their intersection. -/
+@[simp]
+lemma eand_val_def {Ω : Type*} [probability_space Ω] (A B : event Ω) :
+  (A ∩ B).val = A.val ∩ B.val := by refl
+
+/- Event intersection eand is commutative. -/
+lemma eand_comm {Ω : Type*} [probability_space Ω] (A B : event Ω) :
+  (A ∩ B) = (B ∩ A) :=
+begin
+  apply event.eq,
+  simp [set.inter_comm],
+end
+
+/- Event intersection eand is associative. -/
+lemma eand_assoc {Ω : Type*} [probability_space Ω] (A B C : event Ω) :
+  ((A ∩ B) ∩ C) = (A ∩ (B ∩ C)) :=
+begin
+  apply event.eq,
+  simp [set.inter_assoc],
+end
+
+lemma eand_eq_self_of_subset_left {Ω : Type*} [probability_space Ω] (A B : event Ω) :
+  (A.val ⊆ B.val) →
```

Comment:
Use `A <= B`, and then delete this lemma because it is `inf_eq_left`

Reply:
Agreed